packages/trie/src/secure.ts

```diff
@@ -89,8 +89,11 @@ export class SecureTrie extends CheckpointTrie {
    * @param includeCheckpoints - If true and during a checkpoint, the copy will contain the checkpointing metadata and will use the same scratch as underlying db.
    */
   copy(includeCheckpoints = true): SecureTrie {
-    const trie = super.copy(includeCheckpoints)
-    const db = trie.db.copy()
-    return new SecureTrie(db._leveldb, this.root)
+    const db = this.db.copy()
+    const secureTrie = new SecureTrie(db._leveldb, this.root)
+    if (includeCheckpoints && this.isCheckpoint) {
+      secureTrie.db.checkpoints = this.db.checkpoints.slice()
```

Comment:
is this `slice` for a (shallow) copy?

I've never seen `slice` without any params, in ES6 you could use the spread syntax: `[...this.db.checkpoints]`

Reply:
Yep this copies the array. I took this from the existing implementation of `CheckpointTrie.copy` - but I agree it looks a bit confusing. Will improve.